mmengine/analysis/jit_analysis.py

```diff
@@ -0,0 +1,687 @@
+
+import logging
+import typing
+import warnings
+from collections import Counter
+from copy import copy
+from dataclasses import dataclass
+from numbers import Number
+from typing import (Any, Dict, Iterable, Iterator, List, Optional, Set, Tuple,
+                    TypeVar, Union)
+
+import numpy as np
+import torch
+import torch.nn as nn
+from torch import Tensor
+from torch.jit import TracerWarning, _get_trace_graph
+
+from mmengine.logging import print_log
+from .jit_handles import Handle
+
+T = TypeVar('T', bound='JitModelAnalysis')
+
+# Only ignore ops that are technically truly 0 flops:
+# shape-manipulation ops, integer ops, memory copy ops
+_IGNORED_OPS: Set[str] = {
+    'aten::Int',
+    'aten::ScalarImplicit',
+    'aten::__and__',
+    'aten::arange',
+    'aten::bitwise_not',
+    'aten::cat',
+    'aten::chunk',
+    'aten::clamp',
+    'aten::clamp_',
+    'aten::constant_pad_nd',
+    'aten::contiguous',
+    'aten::copy_',
+    'aten::detach',
+    'aten::dropout',
+    'aten::empty',
+    'aten::eq',
+    'aten::expand',
+    'aten::flatten',
+    'aten::floor',
+    'aten::floor_divide',
+    'aten::full',
+    'aten::full_like',
+    'aten::gather',
+    'aten::ge',
+    'aten::gt',
+    'aten::index',
+    'aten::index_put_',
+    'aten::masked_fill',
+    'aten::max',
+    'aten::narrow',
+    'aten::new_empty',
+    'aten::new_full',
+    'aten::new_zeros',
+    'aten::nonzero',
+    'aten::ones',
+    'aten::permute',
+    'aten::relu',
+    'aten::relu_',
+    'aten::remainder',
+    'aten::reshape',
+    'aten::roll',
+    'aten::select',
+    'aten::size',
+    'aten::slice',
+    'aten::split',
+    'aten::split_with_sizes',
+    'aten::squeeze',
+    'aten::stack',
+    'aten::t',
+    'aten::to',
+    'aten::transpose',
+    'aten::type_as',
+    'aten::unbind',
+    'aten::unsqueeze',
+    'aten::unsqueeze_',
+    'aten::view',
+    'aten::zeros',
+    'aten::zeros_like',
+}
+
+
+@dataclass
+class Statistics:
+    """For keeping track of the various model statistics recorded during
+    analysis."""
+
+    counts: 'Dict[str, typing.Counter[str]]'
+    unsupported_ops: 'Dict[str, typing.Counter[str]]'
+    uncalled_mods: 'Set[str]'
+
+
+def _named_modules_with_dup(model: nn.Module,
+                            prefix: str = ''
+                            ) -> Iterable[Tuple[str, nn.Module]]:
+    """The same as `model.named_modules()`, except that it includes duplicated
+    modules that have more than one name."""
+    yield prefix, model
+    for name, module in model._modules.items():
+        if module is None:
+            continue
+        submodule_prefix = prefix + ('.' if prefix else '') + name
+        yield from _named_modules_with_dup(module, submodule_prefix)
+
+
+def _named_modules_without_dup(
+        model: nn.Module) -> Iterator[Tuple[str, nn.Module]]:
+    """Like .named_modules(), but the results are slightly different for some
+    wrapped models."""
```

Comment:
Could you provide a wrapped model which has a different result on this function and `.named_modules()`?